### PR TITLE
Changing date/time datatype conversion so assert_xydata() correctly handles date/time data.

### PR DESCRIPTION
**Pull request by marty-larocque**
This PR changes how date/time data is handled and converted in `get_xydata()` and `assert_xydata()`. Now, timeseries data can be compared using `_compare_time()`. This method accounts for two possible epochs that matplotlib may use. 

This PR addresses issue #169. It should also resolve issue #74.

### CODE REVIEW

**Comment by lwasser** [comment]
fixed the merge conflict!

**Review by nkorinek — CHANGES_REQUESTED** [review]
@ryla5068 Hey! I just installed this locally and was unable to make it run on the week 1 notebook properly. I had data that was identical to the data stored in the PlotTester object, but when I ran `assert_xydata()` it failed saying incorrect data values. Any idea what I may be doing wrong?

**Comment by lwasser** [comment]
@nkorinek  thank you for testing this. can you please try to test this by extracting the data values from the plot as we just discussed in our checkin? let me know if that works.

**Comment by nkorinek** [comment]
@lwasser @ryla5068 Hey! Just to expand on my trouble shooting and found issues, I made a gist showing what failed when I tried different approaches. I hope this helps clear up some of the issues! If you have any questions let me know. https://gist.github.com/nkorinek/0c782d8f8ba59131c1b6d0726f21e89d

**Comment by lwasser** [comment]
@nkorinek thanks!! let's test this in the first notebook but using the plot tester approach that you used for the RS data. that may make our lives easier!!

**Comment by nkorinek** [comment]
@lwasser Hey! So, this data is actually from the first notebook! I just took out the parts that I didn't need for the testing, but it's pulled directly from that notebook. Also, I believe I did test it the way that we handled the RS data, unless I'm misunderstanding something. Cells 5, 6, and 7 in this notebook use a MatPlotCheck object that I created from the axis made by the plot. If this isn't what you had in mind let me know please, thanks!

**Comment by lwasser** [comment]
excellent work @nkorinek !! i just tried exactly what you tried in your gist - testing the data WITHOUT the conversion. i think that is the solution. we remove the xtime parameter and any transformation on the data because it actually works without transforming it. then we just have the instructor grab that xy data from their example expected plot and use it as a comparison.

I think that is the most robust solution which will handle and remove more edge cases than getting into many different types of conversions. We still want to test xy on the x axis, etc but no conversions on the data for time or in general!!

File: matplotcheck/base.py [hunk 1]
```diff
@@ -756,7 +757,7 @@ def assert_no_legend_overlap(self, message="Legends overlap eachother"):
 
     """ BASIC PLOT DATA FUNCTIONS """
 
-    def get_xy(self, points_only=False, xtime=False):
+    def get_xy(self, points_only=False):
```
Comment (marty-larocque) [inline matplotcheck/base.py:760]: I have changed the way that `get_xy()` handles time data. Previously, the docstring stated that if `xtime=True`, then the function would return `datetime.datetime` objects. (Because the data was returned in a dataframe and the way Pandas handles datetime data, it would actualy return `Pandas.timestamp` objects.)

The conversions from matplotlib's datetime datatype to `datetime.datetime` objects is inherently inconsistent, and it made comparisons for datetime data much harder in `assert_xydata()`.

Since `assert_xydata()` was the only function really using the `xtime` flag before, and doesn't use it now, I removed the flag.

Any uses of the `xtime` flag elsewhere in the code just existed for consistency with `assert_xydata()` and have also been removed.

File: matplotcheck/base.py [hunk 2]
```diff
@@ -921,20 +918,127 @@ def assert_xydata(
...
             try:
                 np.testing.assert_array_max_ulp(
-                    np.array(xy_data["y"]), np.array(xy_expected[ycol])
+                    xy_data["y"].values.tolist(),
+                    xy_expected[ycol].values.tolist(),
                 )
             except AssertionError:
                 raise AssertionError(message)
 
+    def _compare_time(self, time_1, time_2):
```
Comment (marty-larocque) [inline matplotcheck/base.py:945]: I created a _compare_time function to handle time comparisons between matplotlib's format and `pandas.timestamp`. This cleans up `assert_xydata()` and allows this code to be reused elsewhere.

File: matplotcheck/tests/test_timeseries_module.py [hunk 3]
```diff
@@ -1,9 +1,152 @@
...
+    return pd.DataFrame(
+        {
+            "time": pd.date_range(start="1/1/1900", periods=100),
+            "A": np.random.randint(0, 100, size=100),
+        }
+    )
+
+
+@pytest.fixture
+def pd_df_timeseries_csv():
```
Comment (marty-larocque) [inline matplotcheck/tests/test_timeseries_module.py:36]: For whatever reason, the datetime data in the homework file was being converted by matplotlib completely differently (i.e. using a different epoch). I have not idea what causes this difference. (It is not because the dates are before 1970. - in other cases, it will still use the 1970 epoch.)

This fixture imports similar data as if it was importing from a CSV, and for whatever reason, it is a reproducible way for matplotlib to use the other epoch.

**Comment by lwasser** [comment]
@ryla5068 so let's pull out tests that we can use from this PR and create a new PR with just the test coverage. We will hold off on implementing conversions for now as they seem fragile and sensitive to changes in matplotlib in the future. Instead we can just pull data from the expected plot itself and use that to compare.
